### PR TITLE
fix: ticks that fall within a discontinuity are no longer removed

### LINKED ISSUE #1601: Discontinuities can result in an axis without any ticks
xScale = fc.scaleDiscontinuous(d3.scaleTime())
            .discontinuityProvider(fc.discontinuitySkipWeekends());

When I change the scale(zoom), I get a bug. Labels and text on the x-scale disappear. Data is an array with a step of one day. The data is sorted. UTC format.

 How to fix?
example

### PR DESCRIPTION
**Pull request by ColinEberhardt**
fixes #1601 #1167 #1464

### CODE REVIEW

File: packages/d3fc-discontinuous-scale/src/tickFilter.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
+import { pairs } from 'd3-array';
+
+function areEqual([a, b]) {
+    if (a.valueOf) {
```
Comment (chrisprice) [inline packages/d3fc-discontinuous-scale/src/tickFilter.js]: Everything in JS has a `valueOf` i.e. [`Object.prototype.valueOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/valueOf) other than `null`/`undefined`. However, they'd cause a `TypeError` here so I guess that's not what you're aiming to catch.

How about a `values.map(d => d?.valueOf())`?

File: .eslintrc.json [hunk 2]
```diff
@@ -31,7 +31,8 @@
     {
       "files": [
         "./packages/d3fc-chart/src/**/*.js",
-        "./packages/d3fc-element/src/**/*.js"
+        "./packages/d3fc-element/src/**/*.js",
+        "./packages/d3fc-discontinuous-scale/src/**/*.js"
```
Comment (chrisprice) [inline .eslintrc.json]: Is this just to support `console.warn`? Maybe it's worth us adding a utility package to wrap this up so we don't need to add every package to this list?

File: packages/d3fc-discontinuous-scale/src/tickFilter.js [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
 export default function tickFilter(ticks, discontinuityProvider) {
-    const discontinuousTicks = [];
-    for (const tick of ticks) {
-        const up = discontinuityProvider.clampUp(tick);
-        const down = discontinuityProvider.clampDown(tick);
-        if (up === down) {
-            discontinuousTicks.push(up);
-        }
+    const discontinuousTicks = ticks.map(discontinuityProvider.clampUp);
+    if (pairs(discontinuousTicks).some(areEqual)) {
```
Comment (chrisprice) [inline packages/d3fc-discontinuous-scale/src/tickFilter.js]: How about `discontinuousTicks.length !== new Set(discontinousTicks.map(d => d?.valueOf())).length`?

(You can drop the new dependency in that case)

**Review by chrisprice — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 2 above)
Comment (ColinEberhardt) [reply .eslintrc.json]: Is your main concern that by moving this package into this particular override section that it now permits all browser globals? How about we just enable console as a global variable across all packages?

https://github.com/d3fc/d3fc/pull/1631/files#diff-df39304d828831c44a2b9f38cd45289cR12

Comment (chrisprice) [reply .eslintrc.json]: It was and that definitely solves it in a nicer way. However, we now risk accidently leaving console logs in the code. I wonder if we should just excuse the individual log lines?

Comment (ColinEberhardt) [reply .eslintrc.json]: We can prevent `console.log` explicitly:

https://github.com/d3fc/d3fc/pull/1631/files#diff-df39304d828831c44a2b9f38cd45289cR22

**Review by chrisprice — APPROVED** [review]

**Pull request merged**